### PR TITLE
[7782] django upgrade

### PR DESCRIPTION
**Pull request by m4ra**
Some issues:

[django 4.0](https://docs.djangoproject.com/en/4.0/releases/4.0/#features-removed-in-4-0) set_language doesn't set the user language in request.session (key _language)
[django-cps](https://github.com/mozilla/django-csp) package has not been updated since 2yrs ago and gives deprecation warning:

> csp/utils.py:105: DeprecationWarning: child-src is deprecated in CSP v3. Use frame-src and worker-src

[django-multiselectfield](https://github.com/goinnn/django-multiselectfield) also appears unsupported/dead and gives error with django > 4.0. Fixed by adding max_length in respected models that use MultiSelectField

### CODE REVIEW

**Comment by goapunk** [comment]
sounds like we maybe should look into getting rid of django-multiselectfield :/

**Comment by m4ra** [comment]
> sounds like we maybe should look into getting rid of django-multiselectfield :/

@goapunk it's fixed by adding max_length.
Bcz if we get rid of this one, we shall also get rid of django-cps.. also not changed since 2yrs ago

File: apps/images/migrations/0006_alter_customrendition_file.py
Comment (m4ra) [inline apps/images/migrations/0006_alter_customrendition_file.py:19]: @goapunk was that a missed one from wagtail upgrade?

Comment (goapunk) [reply apps/images/migrations/0006_alter_customrendition_file.py:19]: no, it isn't created on main. So must be something related to django

File: changelog/7782.md
Comment (goapunk) [inline changelog/7782.md:15]: is this only compatible with wagtail 5.2 or why do we need to wait ?

**Review by goapunk — COMMENTED** [review]
Looks good to me, just some questions

File: changelog/7782.md
Comment (m4ra) [reply changelog/7782.md:15]: yes, current wagtail 5.1.3 compatible to taggit > 5.0

**Review by philli-m — APPROVED** [review]
Nice! quick click around and multi select field adding is all working smooth, couple questions but I think it's good to go, as we wait anyway for @goapunk PR will leave merge to him

File: digitalstrategie/templates/includes/upper_footer.html
Comment (m4ra) [reply digitalstrategie/templates/includes/upper_footer.html]: hmm don't remember why i put following there, but from release 4.0:
> django.views.i18n.set_language() doesn’t set the user language in request.session (key _language).
ref: https://docs.djangoproject.com/en/4.0/releases/4.0/#features-removed-in-4-0
this url only gets added when language is german, but I don't get what's doing, as it is also hidden in the footer.

Comment (m4ra) [reply digitalstrategie/templates/includes/upper_footer.html]: I removed that comment, but can put it back in a more readable language :)

File: changelog/7782.md
Comment (m4ra) [inline changelog/7782.md:15]: @goapunk unfortunaltely 5.2 still only compatible with taggit < 5.0

**Comment by m4ra** [comment]
> some notes:
allowed_hosts needs update in admin once merged, see https://docs.djangoproject.com/en/4.2/releases/4.0/#csrf-trusted-origins-changes

@goapunk so we cannot merge yet bcz admin repo stalk update messed it up?

**Comment by goapunk** [comment]
> > some notes:
> > allowed_hosts needs update in admin once merged, see https://docs.djangoproject.com/en/4.2/releases/4.0/#csrf-trusted-origins-changes
> 
> @goapunk so we cannot merge yet bcz admin repo salt update messed it up?

no, should be ok to merge, these kind of changes still work

File: requirements/prod.txt
Comment (m4ra) [reply requirements/prod.txt:2]: [c]?

File: requirements/base.txt
Comment (m4ra) [reply requirements/base.txt:5]: wagtail

File: requirements/base.txt
Comment (m4ra) [reply requirements/base.txt:7]: Ic, where should I add the comment then?

File: requirements/prod.txt
Comment (goapunk) [reply requirements/prod.txt:2]: see https://www.psycopg.org/psycopg3/docs/basic/install.html#local-installation

File: requirements/base.txt
Comment (goapunk) [reply requirements/base.txt:7]: above the line ? :)

Comment (m4ra) [reply requirements/base.txt:7]: it comes with wagtail, so I will remove it.

**Review by goapunk — APPROVED** [review]
LGTM

**Pull request merged**
